### PR TITLE
disallowNewlineBeforeBlockStatements: add allExcept multiLine exception

### PR DESCRIPTION
**Pull request by ficristo**
This should fix https://github.com/jscs-dev/node-jscs/issues/1956.

### CODE REVIEW

File: lib/rules/disallow-newline-before-block-statements.js [hunk 1]
```diff
@@ -193,11 +256,34 @@ module.exports.prototype = {
...
-                errors.assert.sameLine({
-                    token: prevToken,
-                    nextToken: openingBrace,
-                    message: 'Newline before curly brace for block statement is disallowed'
-                });
+                if (hasMultiLineEx === true) {
+                    var parentNode = node.parentNode;
+                    var parentNextToken = file.getFirstNodeToken(parentNode);
+                    var openingRoundBrace = file.findNextToken(parentNextToken, 'Punctuator', '(');
+
```
Comment (markelog) [inline lib/rules/disallow-newline-before-block-statements.js]: Why there is a newline?

File: lib/rules/disallow-newline-before-block-statements.js [hunk 2]
```diff
@@ -193,11 +256,34 @@ module.exports.prototype = {
...
+                    var closingRoundBrace = file.findPrevToken(openingBrace, 'Punctuator', ')');
+
+                    if (openingRoundBrace && closingRoundBrace &&
```
Comment (markelog) [inline lib/rules/disallow-newline-before-block-statements.js]: There is a lot of nested `if` which usually not a good sign. Would you mind trying [requireEarlyReturn](http://jscs.info/rule/requireEarlyReturn)?

File: lib/rules/disallow-newline-before-block-statements.js [hunk 3]
```diff
@@ -193,11 +256,34 @@ module.exports.prototype = {
...
+                        openingRoundBrace.loc.start.line !== closingRoundBrace.loc.end.line) {
```
Comment (markelog) [inline lib/rules/disallow-newline-before-block-statements.js]: Also, couple of comments would do good here

Comment (ficristo) [inline lib/rules/disallow-newline-before-block-statements.js]: Done.

File: lib/rules/disallow-newline-before-block-statements.js (hunk 2 above)
Comment (ficristo) [inline lib/rules/disallow-newline-before-block-statements.js]: I changed the hasMultiLineEx condition to use it. I think is enough.

**Comment by markelog** [comment]
Sweetsauce!

**Pull request closed without merging**